code/src/components/Album/AlbumArtists.js

```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { Artist } from './Artist';
+
+export const AlbumArtists = ({ artists }) => {
+  const artistElements = artists.map(artist => {
+    return <Artist key={artist.id} data={artist} />;
```

Comment:
Just the Artist component without return statement, since there is no block body?

When it comes to preparing elements outside of JSX I guess I have picked that up from Youtube tutorials. It felt like a good thing at the time to separate the data layer from the presentation layer. But I do understand what you mean. If this is considered best practice within the industry I will follow you recommendation from now on. I'm always open to best practice :)

Reply:
👍

Separating data from presentation is a really good idea, but in this case, I'd argue that this is all presentation - the only difference between this .map and the 'normal' component JSX below is that this one is consuming an array.

Say you needed to do a bunch of filtering and selecting on an array of data which you'll eventually use .map on, I think that's a good case for moving the filtering functions up and then just keep the map in the jsx. Does that make sense? This isn't perhaps the best example, but something like this is what I mean:

```
const filteredArtists = artists
  .filter(artist => artist.rating > 75)
  .map(artist => {
    artist.description = 'Some extra description blah blah'
    return artist
  })

return (
  <div>
    {filteredArtists.map((artist) => (
      <Artist artist={artist} />
    ))}
  </div>
)
```